en/middleware/micrortps.md

```diff
@@ -0,0 +1,338 @@
+# PX4-FastRTPS bridge
+
+
+This bridge adds communication capabilities between a **PX4 Autopilot** and a **Fast RTPS** application through serial ports or
+UDP sockets using **CDR serialization**. The goal is to provide a DDS (Data Distribution Service) interface to PX4. This interface allows sharing information also with the upcoming release of ROS (Robot Operating System), ROS2.
+
+DDS is an standard from the OMG (Object Management Group) providing a real-time pub/sub middleware widely used in aerospace, defense and IoT applications, while in Robotics has been adopted as the middleware for ROS2.
+
+![alt text](res/1_general-white.png)
+
+## Automatic code generation
+
+The support for the functionality is mainly done within three new (automatically generated) code blocks.
+
+-  CDR serialization support is generated for every uORB topic adding a interfaces what for *sensor_combined.msg* topic looks like this:
+
+  ```sh
+  void serialize_sensor_combined(const struct sensor_combined_s *input, char *output, uint32_t *length, struct microCDR *microCDRWriter);
+  void deserialize_sensor_combined(struct sensor_combined_s *output, char *input, struct microCDR *microCDRReader);
+  ```
+
+-  An application to send and receive the CDR serialized data from several topics through a selected UART or selected UDP ports. This is called the client.
+
+![alt text](res/2_trasnmitter-white.png)
+
+-  The other side of the communication is an application between CDR serialized data and the
+ROS2/DDS world (using **Fast RTPS**). An important step in this part is the generation of a idl file for each topic which
+is the translation of the correspondent msg file. For the case of  *sensor_combined* topic it's generated the
+*sensor_combined_.idl* file from *sensor_combined.msg* file. This application is called the agent.
+
+![alt text](res/3_receiver-white.png)
+
+This covers the entire spectrum of communications.
+
+![alt text](res/4_both-white.png)
+
+These pieces of code are generated within the normal PX4 Firmware generation process. Also can be generated under demand calling the script **generate_microRTPS_bridge.py** placed in *Tools* folder, see section below.
+
+### Generating the client and the agent
+
+**NOTE**: Before continue we need to have installed Fast RTPS. Visit its installation
```

Comment:
Should this link be to https://dev.px4.io/en/setup/fast-rtps-installation.html ( ../setup/fast-rtps-installation.md) ?